doc/designs/fips_indicator.md

```diff
@@ -0,0 +1,222 @@
+        return 0;
+    }
+    return 1;
+}
+```
+
+- Existing security check changes
+
+OpenSSL currently does checks that are conditionally based on FIPS config values..
+
+```c
+    if (ossl_securitycheck_enabled(ctx)) {
+       pass = do_some_alg_test(ctx);
+       if (!pass)
+        return 0; /* Produce an error */
+    
+    }
+```
+
+These need to change to work as indicators so the test always runs.. i.e.
+
+```c
+    pass = do_some_alg_test(ctx);
+    // Do code similar to alg_check_approved() above
+```
+
+Notes
+----------
+
+There was discussion related to also having a global config setting that could
+turn off FIPS mode. This will not be added at this stage.
+
+New Changes Required
+--------------------
+
+### key size >= 112 bits
+
+There are a few places where we do not enforce key size that need to be addressed.
+
+- HMAC  Which applies to all algorithms that use HMAC also (e.g. HKDF, SSKDF, KBKDF)
+- CMAC
+- KMAC
+
+### Algorithm Transitions
+
+- DES_EDE3_ECB.  Disallowed for encryption, allowed for legacy decryption
+- DSA.  Keygen and Signing are no longer approved, verify is still allowed.
+- ECDSA B & K curves are deprecated, but still approved according to (IG C.K Resolution 4).\
+  If we chose not to remove them , then we need to check that OSSL_PKEY_PARAM_USE_COFACTOR_ECDH is set for key agreement if the cofactor is not 1.
+- ED25519/ED448 is now approved.
+- X25519/X448 is not approved currently. keygen and keyexchange would also need an indicator if we allow it?
+- RSA encryption(for key agreement/key transport) using PKCSV15 is no longer allowed. (Note that this breaks TLS 1.2 using RSA for KeyAgreement),
+  Padding mode updates required. Check RSA KEM also.
+- RSA signing using PKCS1 is still allowed (i.e. signature uses shaXXXWithRSAEncryption)
+- RSA signing using X931 is no longer allowed. (Still allowed for verification). Check if PSS saltlen needs a indicator (Note FIPS 186-4 Section 5.5 bullet(e). Padding mode updates required in rsa_check_padding(). Check if sha1 is allowed?
+- RSA - (From SP800-131Ar2) RSA >= 2048 is approved for keygen, signatures and key transport. Verification allows 1024 also. Note also that according to the (IG section C.F) that fips 186-2 verification is also allowed (So this may need either testing OR an indicator - it also mentions the modulus size must be 1024 * 256*s). Check that rsa_keygen_pairwise_test() and RSA self tests are all compliant with the above RSA restrictions.
+
+- TLS1_PRF  If we are only trying to support TLS1.2 here then we should remove the tls1.0/1.1 code from the FIPS MODULE.
+
+### Digest Checks
+
+Any algorithms that use a digest need to make sure that the CAVP certificate lists all supported FIPS digests otherwise an indicator is required.
+This applies to the following algorithms:
+
+- TLS_1_3_KDF (Only SHA256 and SHA384 Are allowed due to RFC 8446  Appendix B.4)
+- TLS1_PRF (Only SHA256,SHA384,SHA512 are allowed)
```

Comment:
That not all approved digests are permitted, means an indicator is mandatory.  It could be implicit by erroring with other digests as the DRBGs do.